src/Modal.js

```diff
@@ -61,29 +62,43 @@ export default class ReactModal2 extends React.Component {
     }
   }
 
+  handleBackdropMouseDown = event => {
+    this.backdropMouseDown = !this.modal.contains(event.target)
+  }
+
+  handleBackdropMouseUp = event => {
+    this.backdropMouseUp = !this.modal.contains(event.target)
+  }
+
   handleBackdropClick = () => {
-    if (this.props.closeOnBackdropClick && !this.preventModalClose) {
+    if (
+      this.props.closeOnBackdropClick &&
+      this.backdropMouseDown &&
+      this.backdropMouseUp
```

Comment:
How certain are we that all browsers fire the `click` event _after_ the `mouseUp` event?

Reply:
fairly certain. Only IE 5-8 seem to differ.
https://www.quirksmode.org/dom/events/click.html#t07